cli/integration-test/integration/settings_change_test.clj

```diff
@@ -37,15 +45,14 @@
         :tags [1]}]
       (lsp/await-diagnostics "diagnostics/unused_public_var.clj")))
 
-  (spit current-lsp-config-file new-lsp-config-content)
-
-  (Thread/sleep 1100)
-
-  (lsp/notify! (fixture/did-open-notification "diagnostics/unused_public_var.clj"))
+  (with-revised-lsp-config-content
```

Comment:
Maybe we can use a feature from clojure.test for this purpose, called `clojure.test/use-fixtures`, we use it a lot on the unit-tests, check `clojure-lsp.test-helper/reset-db-after-test`

Reply:
`with-revised-lsp-config-content` does look like it should be a fixture. But I see two problems...
1. Fixtures are applied before or after the deftest as a whole, whereas we want to change the config halfway through a deftest
2. Fixtures are applied around every deftest in a namespace.

We could fix (1) by splitting the tests into two deftests, but because of (2) then both deftests would get the revised config.

I think the only way around both problems would be to create a separate namespace, which seems like it would obscure the intention of these tests. So, I'm inclined to keep the test as is. What do you think?

Reply:
makes sense! no problem keeping it this way